tests/test_heap.py

```diff
@@ -154,13 +162,19 @@ def test_fibonacci_heap_decrease_key(self):
             self.assertTrue(x.key in decrease_data)
 
         # test case 3: Negative
-        for x in fh._iterate(fh.root_list):
-            self.assertFalse(x.key in data) #OBS! Have to re-write to take into account [True True True False True]
+        nodes = self.get_all_nodes(fh, fh.root_list)
+        for i in nodes:
+            self.assertFalse(i.key in [3])
 
         #test case 4: Exception
         with self.assertRaises(Exception):
             fh.decrease_key(fh.find_min(), 8)
-    
+
+        #test case 5: After extract 
+        fh.extract_min_node()
+        fh.decrease_key(fh.root_list.child, 1)
+        self.assertEqual(fh.find_min().key, 1)
+        self.assertTrue(fh.find_min() in fh._iterate(fh.root_list))
```

Comment:
Should also check that `fh.root_list.child` is correct after the cuts. If root_list only had one child, this should be `None`, if it had multiple it should be `fh.root_list.child.right`.